pkg/resources/task.go

```diff
@@ -550,32 +580,53 @@ func ReadTask(withExternalChangesMarking bool) schema.ReadContextFunc {
 			}
 		}
 
-		warehouseId := ""
-		if task.Warehouse != nil {
-			warehouseId = task.Warehouse.Name()
-		}
-
-		errorIntegrationId := ""
-		if task.ErrorIntegration != nil {
-			errorIntegrationId = task.ErrorIntegration.Name()
-		}
-
-		finalizedRootTaskId := ""
-		if task.TaskRelations.FinalizedRootTask != nil {
-			finalizedRootTaskId = task.TaskRelations.FinalizedRootTask.FullyQualifiedName()
-		}
-
 		if errs := errors.Join(
-			d.Set("started", task.State == sdk.TaskStateStarted),
-			d.Set("warehouse", warehouseId),
-			d.Set("schedule", task.Schedule),
+			attributeMappedValueReadIfNotEmptyElse(d, "finalize", task.TaskRelations.FinalizedRootTask, func(finalizedRootTask *sdk.SchemaObjectIdentifier) (string, error) {
+				return finalizedRootTask.FullyQualifiedName(), nil
+			}, nil),
+			attributeMappedValueReadIfNotEmptyElse(d, "error_integration", task.ErrorIntegration, func(errorIntegration *sdk.AccountObjectIdentifier) (string, error) {
+				return errorIntegration.Name(), nil
+			}, nil),
+			attributeMappedValueReadIfNotEmptyElse(d, "warehouse", task.Warehouse, func(warehouse *sdk.AccountObjectIdentifier) (string, error) {
+				return warehouse.Name(), nil
+			}, nil),
+			func() error {
+				upperSchedule := strings.ToUpper(task.Schedule)
+				switch {
+				case strings.Contains(upperSchedule, "USING CRON"):
```

Comment:
this logic is really close to the SDK, let's move it maybe (in the next PR), we can discuss how in the next PR